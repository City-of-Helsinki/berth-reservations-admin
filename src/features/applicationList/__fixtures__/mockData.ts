import { BERTH_APPLICATIONS } from '../__generated__/BERTH_APPLICATIONS';
import { getBerthApplicationData } from '../utils';
import { ApplicationStatus, LeaseStatus } from '../../../@types/__generated__/globalTypes';

export const mockData: BERTH_APPLICATIONS = {
  berthApplications: {
    __typename: 'BerthApplicationNodeConnection',
    count: 3,
    edges: [
      // Full structure
      {
        __typename: 'BerthApplicationNodeEdge',
        node: {
          __typename: 'BerthApplicationNode',
          accessibilityRequired: false,
          berthSwitch: {
            __typename: 'BerthSwitchType',
            berthNumber: '1',
            harbor: 'MOCK-HARBOR-0',
            harborName: 'Testisatama',
            id: 'MOCK-BERTH-SWITCH-0',
            pier: 'A',
            reason: { __typename: 'BerthSwitchReasonType', title: 'Reason' },
          },
          boatLength: 6,
          boatModel: 'Marine',
          boatName: 'Cama la Yano',
          boatDraught: 5,
          boatRegistrationNumber: 'A 12345',
          boatType: '1',
          boatWeight: 5,
          boatWidth: 0,
          createdAt: '2020-07-16',
          customer: { __typename: 'ProfileNode', id: 'MOCK-PROFILE' },
          firstName: 'Matti',
          harborChoices: [{ __typename: 'HarborChoiceType', harbor: '1', harborName: '1', priority: 0 }],
          id: 'MOCK-APPLICATION-0',
          lastName: 'Meikäläinen',
          lease: {
            __typename: 'BerthLeaseNode',
            berth: {
              __typename: 'BerthNode',
              number: '7',
              pier: {
                __typename: 'PierNode',
                properties: {
                  __typename: 'PierProperties',
                  harbor: {
                    __typename: 'HarborNode',
                    id: 'MOCK-HARBOR-0',
                    properties: { __typename: 'HarborProperties', name: 'Testisatama' },
                  },
                  identifier: 'G',
                },
              },
            },
            id: 'MOCK-LEASE-0',
            status: LeaseStatus.OFFERED,
          },
          municipality: 'Helsinki',
          status: ApplicationStatus.PENDING,
        },
      },
      // Most structure and nulled fields
      {
        __typename: 'BerthApplicationNodeEdge',
        node: {
          __typename: 'BerthApplicationNode',
          accessibilityRequired: false,
          berthSwitch: {
            __typename: 'BerthSwitchType',
            berthNumber: '2',
            harbor: 'MOCK-HARBOR-0',
            harborName: 'Testisatama',
            id: 'MOCK-BERTH-SWITCH-1',
            pier: 'A',
            reason: null,
          },
          boatLength: 6,
          boatModel: 'Marine',
          boatName: 'Cama la Yano II',
          boatDraught: 5,
          boatRegistrationNumber: 'B 12345',
          boatType: '1',
          boatWeight: 5,
          boatWidth: 0,
          createdAt: '2020-07-16',
          customer: { __typename: 'ProfileNode', id: 'MOCK-PROFILE' },
          firstName: 'Maija',
          harborChoices: null,
          id: 'MOCK-APPLICATION-1',
          lastName: 'Meikäläinen',
          lease: {
            __typename: 'BerthLeaseNode',
            id: 'MOCK-LEASE-1',
            status: LeaseStatus.OFFERED,
            berth: {
              __typename: 'BerthNode',
              number: '7',
              pier: {
                __typename: 'PierNode',
                properties: {
                  __typename: 'PierProperties',
                  identifier: 'Pier Brosnan',
                  harbor: {
                    __typename: 'HarborNode',
                    id: 'MOCK-HARBOR-1',
                    properties: {
                      __typename: 'HarborProperties',
                      name: null,
                    },
                  },
                },
              },
            },
          },
          municipality: 'Helsinki',
          status: ApplicationStatus.PENDING,
        },
      },
      // Minimum fields
      {
        __typename: 'BerthApplicationNodeEdge',
        node: {
          __typename: 'BerthApplicationNode',
          accessibilityRequired: false,
          berthSwitch: null,
          boatLength: 6,
          boatModel: 'Marine',
          boatName: 'Cama la Yano 3-D',
          boatDraught: 5,
          boatRegistrationNumber: 'C 12345',
          boatType: '1',
          boatWeight: 5,
          boatWidth: 0,
          createdAt: '2020-07-16',
          customer: { __typename: 'ProfileNode', id: 'MOCK-PROFILE-0' },
          firstName: 'Matti',
          harborChoices: [
            { __typename: 'HarborChoiceType', harbor: 'MOCK-HARBOR-0', harborName: 'Testisatama', priority: 0 },
          ],
          id: 'MOCK-APPLICATION-2',
          lastName: 'Möttönen',
          lease: null,
          municipality: 'Helsinki',
          status: ApplicationStatus.PENDING,
        },
      },
    ],
  },
  boatTypes: [{ __typename: 'BoatTypeType', id: '1', name: 'Purjevene' }],
};

export const mockTableData = getBerthApplicationData(mockData);
