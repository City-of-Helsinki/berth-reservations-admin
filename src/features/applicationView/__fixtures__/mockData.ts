import {
  INDIVIDUAL_APPLICATION,
  INDIVIDUAL_APPLICATION_additionalProducts as ADDITIONAL_PRODUCTS,
  INDIVIDUAL_APPLICATION_berthApplication as BERTH_APPLICATION,
  INDIVIDUAL_APPLICATION_berthApplication_berthSwitch as BERTH_SWITCH,
  INDIVIDUAL_APPLICATION_berthApplication_customer as CUSTOMER,
  INDIVIDUAL_APPLICATION_berthApplication_lease as LEASE,
  INDIVIDUAL_APPLICATION_boatTypes as BOAT_TYPES,
} from '../__generated__/INDIVIDUAL_APPLICATION';
import {
  AdditionalProductType,
  ApplicationStatus,
  BerthApplicationLanguage,
  BerthMooringType,
  CustomerGroup,
  InvoicingType,
  Language,
  PeriodType,
  PriceUnits,
  ProductServiceType,
} from '../../../@types/__generated__/globalTypes';

export const mockCustomer: CUSTOMER = {
  __typename: 'ProfileNode',
  comment: 'Comment',
  firstName: 'Testi',
  id: 'MOCK-CUSTOMER',
  invoicingType: InvoicingType.DIGITAL_INVOICE,
  language: Language.FINNISH,
  lastName: 'Testinen',
  organization: null,
  primaryAddress: { __typename: 'AddressNode', address: 'Testikatu 1', city: 'Helsinki', postalCode: '00100' },
  primaryEmail: { __typename: 'EmailNode', email: 'test@example.com' },
  primaryPhone: { __typename: 'PhoneNode', phone: '+358 00 000 0000' },
  customerGroup: CustomerGroup.PRIVATE,
};

export const mockBerthSwitch: BERTH_SWITCH = {
  __typename: 'BerthSwitchType',
  berthNumber: '1',
  harbor: 'MOCK-HARBOR',
  harborName: 'Test Harbor',
  id: 'MOCK-BERTH-SWITCH',
  pier: 'A',
  reason: { __typename: 'BerthSwitchReasonType', title: 'Reason' },
};

export const mockLease: LEASE = {
  __typename: 'BerthLeaseNode',
  berth: {
    __typename: 'BerthNode',
    comment: '',
    depth: null,
    isAccessible: null,
    length: 5,
    mooringType: BerthMooringType.DINGHY_PLACE,
    number: 1,
    pier: {
      __typename: 'PierNode',
      properties: {
        __typename: 'PierProperties',
        electricity: false,
        gate: false,
        harbor: {
          __typename: 'HarborNode',
          id: 'MOCK-HARBOR',
          properties: { __typename: 'HarborProperties', name: 'Test Harbor' },
        },
        identifier: 'A',
        lighting: false,
        mooring: false,
        wasteCollection: false,
        water: false,
      },
    },
    width: 0,
  },
  id: 'MOCK-LEASE',
  order: null,
};

const additionalProducts: ADDITIONAL_PRODUCTS = {
  __typename: 'AdditionalProductNodeConnection',
  edges: [
    {
      __typename: 'AdditionalProductNodeEdge',
      node: {
        __typename: 'AdditionalProductNode',
        id: 'MOCK-ADDITIONAL-PRODUCT-NODE',
        period: PeriodType.MONTH,
        priceUnit: PriceUnits.AMOUNT,
        priceValue: '25.00',
        productType: AdditionalProductType.OPTIONAL_SERVICE,
        service: ProductServiceType.DINGHY_PLACE,
      },
    },
  ],
};

const berthApplication: BERTH_APPLICATION = {
  __typename: 'BerthApplicationNode',
  accessibilityRequired: false,
  address: 'Testikatu 1',
  berthSwitch: null,
  boatDraught: 1,
  boatLength: 1,
  boatModel: 'Marine',
  boatName: 'Cama la Yano',
  boatRegistrationNumber: '1234',
  boatType: '7',
  boatWeight: 120,
  boatWidth: 1,
  businessId: '',
  companyName: '',
  createdAt: '2020-07-10T08:04:36.062819+00:00',
  customer: null,
  email: 'test@example.com',
  firstName: 'Testi',
  harborChoices: [
    { harbor: 'MOCK-HARBOR-B', priority: 1, harborName: 'Saukonpaaden venesatama', __typename: 'HarborChoiceType' },
  ],
  id: 'MOCK-APPLICATION',
  language: BerthApplicationLanguage.FI,
  lastName: 'Testinen',
  lease: null,
  municipality: 'Helsinki',
  phoneNumber: '+358 00 000 0000',
  status: ApplicationStatus.PENDING,
  zipCode: '00100',
  acceptBoatingNewsletter: true,
  acceptFitnessNews: false,
  acceptLibraryNews: false,
  acceptOtherCultureNews: true,
  applicationCode: '1234',
};

const boatTypes: BOAT_TYPES[] = [
  { id: '1', name: 'Jollavene', __typename: 'BoatTypeType' },
  { id: '2', name: 'Soutuvene', __typename: 'BoatTypeType' },
  { id: '3', name: 'Perämoottorivene', __typename: 'BoatTypeType' },
  { id: '4', name: 'Sisäperämoottorivene', __typename: 'BoatTypeType' },
  { id: '5', name: 'Keskimoottorivene', __typename: 'BoatTypeType' },
  { id: '6', name: 'Purjevene / moottoripursi', __typename: 'BoatTypeType' },
  { id: '7', name: 'Troolari', __typename: 'BoatTypeType' },
  { id: '8', name: 'Suuri alus (yli 20t)', __typename: 'BoatTypeType' },
];

export const mockApplication: INDIVIDUAL_APPLICATION = {
  __typename: 'Query',
  additionalProducts: additionalProducts,
  berthApplication: berthApplication,
  boatTypes: boatTypes,
};
