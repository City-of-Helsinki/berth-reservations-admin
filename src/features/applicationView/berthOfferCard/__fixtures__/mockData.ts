import { BerthLease_lease as BerthLease } from '../__generated__/BerthLease';
import {
  AdditionalProductType,
  BerthMooringType,
  LeaseStatus,
  ProductServiceType,
} from '../../../../@types/__generated__/globalTypes';

export const lease: BerthLease = {
  id: '165023f5-5b74-4656-9315-d4e1a22f111d',
  status: LeaseStatus.DRAFTED,
  berth: {
    depth: null,
    length: 6,
    mooringType: BerthMooringType.SINGLE_SLIP_PLACE,
    width: 2.5,
    comment: '',
    isAccessible: null,
    number: '1',
    pier: {
      properties: {
        identifier: 'C',
        electricity: true,
        gate: true,
        lighting: true,
        mooring: true,
        wasteCollection: true,
        water: true,
        harbor: {
          id: 'b55da996-cf00-4b52-9aa0-6b13fd8d806b',
          properties: {
            name: 'Aurinkolahden venesatama (Aurinkoranta)',
            __typename: 'HarborProperties',
          },
          __typename: 'HarborNode',
        },
        __typename: 'PierProperties',
      },
      __typename: 'PierNode',
    },
    __typename: 'BerthNode',
  },
  customer: {
    __typename: 'ProfileNode',
    id: 'MOCK-CUSTOMER',
    primaryEmail: {
      __typename: 'EmailNode',
      id: 'MOCK-EMAIL',
      email: 'test@example.com',
    },
  },
  order: {
    id: '07f65f54-4a25-47fd-9eea-d9fdddfa46f9',
    orderNumber: '123456',
    price: '96.67',
    totalPrice: '256.70',
    orderLines: {
      edges: [
        {
          node: {
            id: '9b97b9bc-bba5-46d9-aae2-86e8686012ef',
            price: '20.50',
            product: {
              id: 'dc7f8158-a4cb-49a9-9368-e586f224fdb5',
              service: ProductServiceType.ELECTRICITY,
              productType: AdditionalProductType.FIXED_SERVICE,
              __typename: 'AdditionalProductNode',
            },
            __typename: 'OrderLineNode',
          },
          __typename: 'OrderLineNodeEdge',
        },
        {
          node: {
            id: 'ea2ef175-f790-45a0-b16f-9e5a98340691',
            price: '19.33',
            product: {
              id: '2e55f605-061e-4299-814d-bf2cd53aa8ce',
              service: ProductServiceType.WATER,
              productType: AdditionalProductType.FIXED_SERVICE,
              __typename: 'AdditionalProductNode',
            },
            __typename: 'OrderLineNode',
          },
          __typename: 'OrderLineNodeEdge',
        },
        {
          node: {
            id: '8ba4de61-63cc-49fc-afe2-26068f3c8e4d',
            price: '2.42',
            product: {
              id: '3a2e3a2f-2782-40e9-a139-b9a4d4a36c3b',
              service: ProductServiceType.GATE,
              productType: AdditionalProductType.FIXED_SERVICE,
              __typename: 'AdditionalProductNode',
            },
            __typename: 'OrderLineNode',
          },
          __typename: 'OrderLineNodeEdge',
        },
        {
          node: {
            id: '59c768c1-f371-4178-afda-1e16d2c725b0',
            price: '8.33',
            product: {
              id: '762da6e9-89be-43bc-b097-53dff4ed8415',
              service: ProductServiceType.MOORING,
              productType: AdditionalProductType.FIXED_SERVICE,
              __typename: 'AdditionalProductNode',
            },
            __typename: 'OrderLineNode',
          },
          __typename: 'OrderLineNodeEdge',
        },
        {
          node: {
            id: '6d910d8e-80b6-4486-aa85-4958bcd98dd5',
            price: '4.17',
            product: {
              id: '34c5b6bc-7eb8-457e-bce8-8ba3a07b9057',
              service: ProductServiceType.WASTE_COLLECTION,
              productType: AdditionalProductType.FIXED_SERVICE,
              __typename: 'AdditionalProductNode',
            },
            __typename: 'OrderLineNode',
          },
          __typename: 'OrderLineNodeEdge',
        },
        {
          node: {
            id: 'ab7cf1e5-92b5-42ad-844a-e2d577d30baf',
            price: '40.28',
            product: {
              id: '650dbfcb-8f10-46b6-af64-197b77686a5b',
              service: ProductServiceType.LIGHTING,
              productType: AdditionalProductType.FIXED_SERVICE,
              __typename: 'AdditionalProductNode',
            },
            __typename: 'OrderLineNode',
          },
          __typename: 'OrderLineNodeEdge',
        },
        {
          node: {
            id: '244a7af9-6f91-4f08-9738-82ef06a2e790',
            price: '65.00',
            product: {
              id: '5194dd14-0226-4603-808c-831cfbec72de',
              service: ProductServiceType.DINGHY_PLACE,
              productType: AdditionalProductType.OPTIONAL_SERVICE,
              __typename: 'AdditionalProductNode',
            },
            __typename: 'OrderLineNode',
          },
          __typename: 'OrderLineNodeEdge',
        },
      ],
      __typename: 'OrderLineNodeConnection',
    },
    __typename: 'OrderNode',
  },
  __typename: 'BerthLeaseNode',
};
