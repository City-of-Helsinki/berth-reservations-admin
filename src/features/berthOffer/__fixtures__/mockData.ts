import { BERTH_OFFER } from '../__generated__/BERTH_OFFER';
import { ApplicationStatus, BerthMooringType, LeaseStatus } from '../../../@types/__generated__/globalTypes';
import { SWITCH_BERTH_LEASE } from '../__generated__/SWITCH_BERTH_LEASE';

export const berthOfferQueryData: BERTH_OFFER = {
  berthApplication: {
    __typename: 'BerthApplicationNode',
    id: '1',
    createdAt: null,
    status: ApplicationStatus.PENDING,
    berthSwitch: null,
    customer: {
      __typename: 'ProfileNode',
      id: '1',
    },
    boatType: 'rowboat',
    boatRegistrationNumber: '1234',
    boatName: 'test',
    boatModel: 'test',
    boatWidth: 1,
    boatLength: 1,
    boatDraught: 1,
    boatWeight: 1,
  },
  boatTypes: [
    {
      __typename: 'BoatTypeType',
      id: 'rowboat',
      name: 'rowboat',
    },
  ],
  harbor: {
    __typename: 'HarborNode',
    id: 'SGFyYm9yTm9kZTpmN2M2YTQwZjAtOWViMi0zZjgyMTI0YjY0OGI=',
    properties: {
      __typename: 'HarborProperties',
      name: 'Pajalahden venesatama (Meripuistotie) / Venesatama',
      streetAddress: 'Meripuistotie 1a',
      zipCode: '00210',
      municipality: 'Helsinki',
      imageFile: 'https://venepaikat-api.test.hel.ninja/media/harbors/9a8d8313-eaa2-47d2-8f2d-2bb9893f9bc7/41359.jpg',
      maxWidth: 4,
      servicemapId: '41359',
      numberOfPlaces: 4,
      numberOfFreePlaces: 2,
      piers: {
        __typename: 'PierNodeConnection',
        edges: [
          {
            __typename: 'PierNodeEdge',
            node: {
              __typename: 'PierNode',
              id: '123456',
              properties: {
                __typename: 'PierProperties',
                identifier: 'pier',
                electricity: true,
                gate: true,
                lighting: true,
                wasteCollection: true,
                water: true,
                berths: {
                  __typename: 'BerthNodeConnection',
                  edges: [
                    {
                      __typename: 'BerthNodeEdge',
                      node: {
                        __typename: 'BerthNode',
                        comment: 'No comments',
                        depth: 5,
                        id: '1',
                        isAccessible: true,
                        isActive: true,
                        leases: {
                          __typename: 'BerthLeaseNodeConnection',
                          edges: [
                            {
                              __typename: 'BerthLeaseNodeEdge',
                              node: {
                                __typename: 'BerthLeaseNode',
                                status: LeaseStatus.PAID,
                                startDate: null,
                                endDate: null,
                                isActive: true,
                                customer: {
                                  __typename: 'ProfileNode',
                                  id: '1',
                                },
                              },
                            },
                          ],
                        },
                        length: 10,
                        mooringType: BerthMooringType.DINGHY_PLACE,
                        number: '1',
                        width: 5,
                      },
                    },
                  ],
                },
              },
            },
          },
        ],
      },
    },
  },
};

export const mockSwitchBerthLease: SWITCH_BERTH_LEASE = {
  berthLease: {
    __typename: 'BerthLeaseNode',
    boat: {
      __typename: 'BoatNode',
      boatType: { id: '2', name: 'Soutuvene', __typename: 'BoatTypeType' },
      draught: 0.5,
      id: 'MOCK-BOAT',
      length: 3,
      model: 'MODEL',
      name: 'BOAT',
      registrationNumber: '',
      weight: 100,
      width: 1,
    },
    id: 'MOCK-LEASE',
  },
};
