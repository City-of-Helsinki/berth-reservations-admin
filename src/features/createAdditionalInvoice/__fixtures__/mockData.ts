import {
  AdditionalProductTaxEnum,
  AdditionalProductType,
  PeriodType,
  PriceUnits,
  ProductServiceType,
} from '../../../@types/__generated__/globalTypes';
import { ADDITIONAL_SERVICES } from '../__generated__/ADDITIONAL_SERVICES';
import { CREATE_ADDITIONAL_INVOICE_createAdditionalProductOrder_order as CreateAddtionalInvoiceOrder } from '../__generated__/CREATE_ADDITIONAL_INVOICE';

export const additionalservices: ADDITIONAL_SERVICES = {
  optionalProducts: {
    __typename: 'AdditionalProductNodeConnection',
    edges: [
      {
        __typename: 'AdditionalProductNodeEdge',
        node: {
          __typename: 'AdditionalProductNode',
          id: 'a8400b59-534c-4934-b643-083a4273ca1a',
          service: ProductServiceType.DINGHY_PLACE,
          priceValue: 10,
          priceUnit: PriceUnits.AMOUNT,
          period: PeriodType.SEASON,
          productType: AdditionalProductType.OPTIONAL_SERVICE,
          taxPercentage: AdditionalProductTaxEnum.TAX_24_00,
        },
      },
      {
        __typename: 'AdditionalProductNodeEdge',
        node: {
          __typename: 'AdditionalProductNode',
          id: 'b8400b59-534c-4934-b643-083a4273ca1a',
          service: ProductServiceType.STORAGE_ON_ICE,
          priceValue: 16,
          priceUnit: PriceUnits.AMOUNT,
          period: PeriodType.SEASON,
          productType: AdditionalProductType.OPTIONAL_SERVICE,
          taxPercentage: AdditionalProductTaxEnum.TAX_24_00,
        },
      },
      {
        __typename: 'AdditionalProductNodeEdge',
        node: {
          __typename: 'AdditionalProductNode',
          id: 'c8400b59-534c-4934-b643-083a4273ca1a',
          service: ProductServiceType.STORAGE_ON_ICE,
          priceValue: 10,
          priceUnit: PriceUnits.AMOUNT,
          period: PeriodType.MONTH,
          productType: AdditionalProductType.OPTIONAL_SERVICE,
          taxPercentage: AdditionalProductTaxEnum.TAX_24_00,
        },
      },
    ],
  },
};

export const createdOrderMock: CreateAddtionalInvoiceOrder = {
  __typename: 'OrderNode',
  id: 'T3JkZXJOb2RlOmYwOWZkNWNjLTc3YzUtNGE2ZS1iMGMxLWI3ZDczMDNkZTk1Yw==',
  price: '0.00',
  totalPrice: '138.00',
  lease: {
    id: 'QmVydGhMZWFzZU5vZGU6N2Y1NWFkNDktODljOC00YzBjLWI4MzktZTI4MDE1N2Q5ZWI3',
    startDate: '2021-06-10',
    endDate: '2021-09-14',
    berth: {
      pier: {
        properties: {
          harbor: {
            id: 'SGFyYm9yTm9kZTowODg1ZjUwMi1kNTY2LTQ4MzItYTc4Mi1iMzEzNzU3MmJiYTQ=',
            properties: { name: 'Hietalahdenallas / Venesatama', __typename: 'HarborProperties' },
            __typename: 'HarborNode',
          },
          __typename: 'PierProperties',
        },
        __typename: 'PierNode',
      },
      __typename: 'BerthNode',
    },
    __typename: 'BerthLeaseNode',
  },
  orderLines: {
    edges: [
      {
        node: {
          product: {
            id: 'QWRkaXRpb25hbFByb2R1Y3ROb2RlOjY3MTRhMmNjLTQyYjgtNDMxNi1hZjMyLTVhZTZmMDMxYTM3MQ==',
            service: ProductServiceType.STORAGE_ON_ICE,
            __typename: 'AdditionalProductNode',
          },
          __typename: 'OrderLineNode',
        },
        __typename: 'OrderLineNodeEdge',
      },
    ],
    __typename: 'OrderLineNodeConnection',
  },
};
