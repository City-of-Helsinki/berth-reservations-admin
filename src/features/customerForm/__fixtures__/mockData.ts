import { CUSTOMER_FORM_profile as PROFILE } from '../__generated__/CUSTOMER_FORM';
import { CustomerGroup, InvoicingType, Language } from '../../../@types/__generated__/globalTypes';
import { CustomerFormIdentifiers, CustomerFormValues } from '../types';

export const mockPrivateCustomerProfile: PROFILE = {
  __typename: 'ProfileNode',
  comment: '',
  customerGroup: CustomerGroup.PRIVATE,
  firstName: 'Risto Heikki',
  id: 'MOCK-PROFILE',
  language: Language.FINNISH,
  lastName: 'Ryti',
  organization: null,
  invoicingType: InvoicingType.ONLINE_PAYMENT,
  primaryAddress: {
    __typename: 'AddressNode',
    address: 'Mariankatu 2',
    city: 'Helsinki',
    id: 'MOCK-ADDRESS',
    postalCode: '00170',
  },
  primaryEmail: {
    __typename: 'EmailNode',
    email: 'test@example.com',
    id: 'MOCK-EMAIL',
  },
  primaryPhone: {
    __typename: 'PhoneNode',
    id: 'MOCK-PHONE',
    phone: '+358 00 000 0000',
  },
};

export const mockOrganizationCustomerProfile: PROFILE = {
  __typename: 'ProfileNode',
  comment: '',
  customerGroup: CustomerGroup.COMPANY,
  firstName: 'Ky√∂sti',
  id: 'MOCK-PROFILE',
  language: Language.FINNISH,
  lastName: 'Kallio',
  invoicingType: InvoicingType.ONLINE_PAYMENT,
  organization: {
    __typename: 'OrganizationNode',
    address: 'Mariankatu 2',
    businessId: '1234567-8',
    city: 'Helsinki',
    id: 'MOCK-ORGANIZATION',
    name: 'Suomi Oy',
    postalCode: '00170',
  },
  primaryAddress: null,
  primaryEmail: {
    __typename: 'EmailNode',
    email: 'test@example.com',
    id: 'MOCK-EMAIL',
  },
  primaryPhone: {
    __typename: 'PhoneNode',
    id: 'MOCK-PHONE',
    phone: '+358 00 000 0000',
  },
};

export const mockPrivateCustomerFormValues: CustomerFormValues = {
  address: 'Mariankatu 2',
  businessId: '',
  city: 'Helsinki',
  comment: '',
  customerGroup: CustomerGroup.PRIVATE,
  email: 'test@example.com',
  firstName: 'Pehr Evind',
  lastName: 'Svinhufvud',
  organizationName: '',
  phone: '+358 00 000 0000',
  postalCode: '00170',
};

export const mockPrivateCustomerFormIdentifiers: CustomerFormIdentifiers = {
  id: 'MOCK-PROFILE',
  initialCustomerGroup: CustomerGroup.PRIVATE,
  primaryAddressId: 'MOCK-ADDRESS',
  primaryEmailId: 'MOCK-EMAIL',
  primaryPhoneId: 'MOCK-PHONE',
};

export const mockOrganizationCustomerFormValues: CustomerFormValues = {
  address: 'Mariankatu 2',
  businessId: '1234567-8',
  city: 'Helsinki',
  comment: '',
  customerGroup: CustomerGroup.COMPANY,
  email: 'test@example.com',
  firstName: 'Lauri Kristian',
  lastName: 'Relander',
  organizationName: 'Suomi Oy',
  phone: '+358 00 000 0000',
  postalCode: '00170',
};

export const mockOrganizationCustomerFormIdentifiers: CustomerFormIdentifiers = {
  id: 'MOCK-PROFILE',
  initialCustomerGroup: CustomerGroup.COMPANY,
  primaryAddressId: 'MOCK-ADDRESS',
  primaryEmailId: 'MOCK-EMAIL',
  primaryPhoneId: 'MOCK-PHONE',
};
