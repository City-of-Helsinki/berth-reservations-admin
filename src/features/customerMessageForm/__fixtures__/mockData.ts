import { NOTIFICATION_TEMPLATES } from '../__generated__/NOTIFICATION_TEMPLATES';
import { NotificationTemplateLanguage } from '../../../@types/__generated__/globalTypes';

export const mockHtml =
  // eslint-disable-next-line max-len
  '<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"><html xmlns="http://www.w3.org/1999/xhtml" lang="fi" xml:lang="fi"><head><meta http-equiv="Content-Type" content="text/html; charset=utf-8"><meta name="viewport" content="width=device-width"><title>Test HTML</title></head><body><h1>Test HTML</h1><p>Test HTML</p></body></html>';

export const mockData: NOTIFICATION_TEMPLATES = {
  notificationTemplates: {
    edges: [
      {
        node: {
          id: 'Tm90aWZpY2F0aW9uVGVtcGxhdGVOb2RlOjE=',
          preview: mockHtml,
          translations: [
            {
              bodyHtml: mockHtml,
              bodyText: '',
              languageCode: NotificationTemplateLanguage.EN,
              preview: mockHtml,
              subject: 'Confirmation: Berth application',
              __typename: 'NotificationTranslationType',
            },
            {
              bodyHtml: mockHtml,
              bodyText: '',
              languageCode: NotificationTemplateLanguage.SV,
              preview: mockHtml,
              subject: 'Bekräftelse: båtplatsansökning',
              __typename: 'NotificationTranslationType',
            },
            {
              bodyHtml: mockHtml,
              bodyText: '',
              languageCode: NotificationTemplateLanguage.FI,
              preview: mockHtml,
              subject: 'Vahvistus: venepaikan hakemus',
              __typename: 'NotificationTranslationType',
            },
          ],
          type: 'berth_application_created',
          __typename: 'NotificationTemplateNode',
        },
        __typename: 'NotificationTemplateNodeEdge',
      },
      {
        node: {
          id: 'Tm90aWZpY2F0aW9uVGVtcGxhdGVOb2RlOjI=',
          preview: mockHtml,
          translations: [
            {
              bodyHtml: mockHtml,
              bodyText: '',
              languageCode: NotificationTemplateLanguage.FI,
              preview: mockHtml,
              subject: 'Vahvistus: Veneen talvisäilytyspaikan hakemus',
              __typename: 'NotificationTranslationType',
            },
            {
              bodyHtml: mockHtml,
              bodyText: '',
              languageCode: NotificationTemplateLanguage.SV,
              preview: mockHtml,
              subject: 'Bekräftelse: ansökan om vinteruppläggningsplats',
              __typename: 'NotificationTranslationType',
            },
            {
              bodyHtml: mockHtml,
              bodyText: '',
              languageCode: NotificationTemplateLanguage.EN,
              preview: mockHtml,
              subject: 'Confirmation: boat winter storage application',
              __typename: 'NotificationTranslationType',
            },
          ],
          type: 'winter_storage_application_created',
          __typename: 'NotificationTemplateNode',
        },
        __typename: 'NotificationTemplateNodeEdge',
      },
    ],
    __typename: 'NotificationTemplateNodeConnection',
  },
};
