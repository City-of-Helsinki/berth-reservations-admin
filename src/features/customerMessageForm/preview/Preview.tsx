import React from 'react';
import styles from './preview.module.scss';

export interface PreviewProps {
  html: string;
}

const Preview = ({ html }: PreviewProps) => {
  return (
    <div className={styles.preview}>
      <iframe title="PreviewModal" className={styles.iframe} src={'data:text/html,' + encodeURIComponent(html)} />
    </div>
  );
};

export default Preview;
