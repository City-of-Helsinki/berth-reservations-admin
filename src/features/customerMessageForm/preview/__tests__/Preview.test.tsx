import React from 'react';
import { shallow } from 'enzyme';

import Preview, { PreviewProps } from '../Preview';
import { mockHtml } from '../../__fixtures__/mockData';

const mockProps: PreviewProps = {
  html: mockHtml,
};

describe('Preview', () => {
  const getWrapper = (props?: Partial<PreviewProps>) => shallow(<Preview {...mockProps} {...props} />);

  it('renders normally', () => {
    const wrapper = getWrapper();

    expect(wrapper.render()).toMatchSnapshot();
  });
});
