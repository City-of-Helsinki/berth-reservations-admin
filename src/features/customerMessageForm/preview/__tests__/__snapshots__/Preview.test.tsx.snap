// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview renders normally 1`] = `
<div
  class="preview"
>
  <iframe
    class="iframe"
    src="data:text/html,%3C!DOCTYPE%20html%20PUBLIC%20%22-%2F%2FW3C%2F%2FDTD%20XHTML%201.0%20Transitional%2F%2FEN%22%20%22http%3A%2F%2Fwww.w3.org%2FTR%2Fxhtml1%2FDTD%2Fxhtml1-transitional.dtd%22%3E%3Chtml%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F1999%2Fxhtml%22%20lang%3D%22fi%22%20xml%3Alang%3D%22fi%22%3E%3Chead%3E%3Cmeta%20http-equiv%3D%22Content-Type%22%20content%3D%22text%2Fhtml%3B%20charset%3Dutf-8%22%3E%3Cmeta%20name%3D%22viewport%22%20content%3D%22width%3Ddevice-width%22%3E%3Ctitle%3ETest%20HTML%3C%2Ftitle%3E%3C%2Fhead%3E%3Cbody%3E%3Ch1%3ETest%20HTML%3C%2Fh1%3E%3Cp%3ETest%20HTML%3C%2Fp%3E%3C%2Fbody%3E%3C%2Fhtml%3E"
    title="PreviewModal"
  />
  <div
    class="buttons"
  >
    <button
      class="Button-module_button__3wgee button_hds-button__2A0je Button-module_secondary__399Vd button_hds-button--secondary__1NOWS Button-module_theme-coat__2xA5D button_hds-button--theme-coat__12cO0"
      type="button"
    >
      <span
        class="Button-module_label__2zMLC button_hds-button__label__2EQa-"
      >
        Peruuta
      </span>
    </button>
  </div>
</div>
`;
